tests/run-tests.sh

```diff
@@ -1,7 +1,10 @@
 #!/bin/bash
+set -eo pipefail
 cd $(dirname $0)
 
+pip install -r ./requirements.txt
```

Comment:
consider separate out installing requirements in run-test.sh since it only needs to do once. Suggest to do this in "Install dependencies for tests" of the workflow_tests.yml.